### PR TITLE
collection - generic metadata

### PR DESCRIPTION
**Pull request by phweyland**
Is part of the work to turn metadata generic.

- turn metadata generic in collection
- add entry 'none' for metadata collections
- fix collection not refreshed when metadata change
- make "clear metadata" keep internal and hidden metadata

@wpferguson, I may have disturbed lua connection to collect module (collect.c). Could you check please this is not the case ? Thanks.

### CODE REVIEW

**Comment by wpferguson** [comment]
@phweyland I've downloaded the PR and I'll test all the scripts that use darktable.gui.libs.collect... and make sure they still work

**Comment by wpferguson** [comment]
@phweyland I've downloaded the PR and I'll test all the scripts that use darktable.gui.libs.collect... and make sure they still work

**Comment by phweyland** [comment]
> Is part of the work to turn metadata generic.
> 
>     * turn metadata generic in collection
> 
>     * add entry 'none' for metadata collections
> 
>     * fix collection not refreshed when metadata change
> 
>     * make "clear metadata" keep internal and hidden metadata
> 
> 
> @wpferguson, I may have disturbed lua connection to collect module (collect.c). Could you check please this is not the case ? Thanks.

**Comment by wpferguson** [comment]
@phweyland all existing scripts that use darktable.gui.libs.collect... still work fine.

**Comment by phweyland** [comment]
>  all existing scripts that use darktable.gui.libs.collect... still work fine.

Thanks so much @wpfergusson !

File: src/libs/metadata.c [hunk 1]
```diff
@@ -243,6 +243,7 @@ static void write_metadata(dt_lib_module_t *self)
   g_list_free(key_value);
 
   dt_control_signal_raise(darktable.signals, DT_SIGNAL_MOUSE_OVER_IMAGE_CHANGE);
+  dt_control_signal_raise(darktable.signals, DT_SIGNAL_METADATA_CHANGED, 2);
```
Comment (TurboGit) [inline src/libs/metadata.c]: we have 2 here...

File: src/libs/metadata.c [hunk 2]
```diff
@@ -475,21 +476,30 @@ static void config_button_clicked(GtkButton *button, dt_lib_module_t *self)
...
+        }
         g_free(setting);
         setting = dt_util_dstrcat(NULL, "plugins/lighttable/metadata/%s_private", name[i]);
         const gboolean private_nv = gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(private[i]));
         if(private_nv != private_v[i]) dt_conf_set_bool(setting, private_nv);
         g_free(setting);
       }
     }
+    if(meta_signal)
+      dt_control_signal_raise(darktable.signals, DT_SIGNAL_METADATA_CHANGED, meta_remove ? 1 : 0);
```
Comment (TurboGit) [inline src/libs/metadata.c]: and 0 or 1 here.

Those values seems to control something, I would certainly prefer having an enum with proper names.

File: src/libs/collect.c [hunk 3]
```diff
@@ -2074,14 +2076,70 @@ static void tag_changed(gpointer instance, gpointer self)
...
     d->rule[d->active_rule].typing = FALSE;
     _lib_collect_gui_update(self);
   }
 }
 
+static void metadata_changed(gpointer instance, int type, gpointer self)
+{
+  dt_lib_module_t *dm = (dt_lib_module_t *)self;
+  dt_lib_collect_t *d = (dt_lib_collect_t *)dm->data;
+  if(type != 2)
```
Comment (TurboGit) [inline src/libs/collect.c]: And here type 0, 1, 2 should also be using the enum I'm proposing to introduce.

File: src/libs/metadata.c (hunk 2 above)
Comment (phweyland) [reply src/libs/metadata.c]: Yes, sure !

**Comment by TurboGit** [comment]
Ok, works for me. Thanks.

**Pull request merged**